### ISSUE TITLE
Is it totally necesary to copy each frame?

### ISSUE THREAD

**Issue by EquipoP** [carykh/jumpcutter#49]
When working with big files (about two hour long videos) the time-altered frame saving takes a very long while, and I was wondering if it wasn't possible to calculate which frames will disappear and just delete the frames you won't need and use the ones remaining as input for ffmeg. Am I mising something?

**Comment by egrinant** [comment]
Had the same thought and tried with ffmpeg params - vf select and - af aselect with "between(n, chunk[0],chunk[1])" on chunks with chunk[2] equal to 1. The problem is that the generated command exceeds Windows cmd's limit of 8192 characters. I've tried also to generate a txt file with the parameters but couldn't make it work. Be aware that this technique will cut the video and the audio sequence, but with the original approach is not to cut but to speed up the silenced parts, giving a more natural feeling.

**Comment by egrinant** [comment]
I found a script that does it much faster (by a lot) and doesnt have to extract each frame, saving time and space. Also it doesn't seem to have any A/V desync problems (an issue no one has reported yet).

https://github.com/excitoon/video-remove-silence

I am not trying to keep people away from carykh's project but to point to an alternative and a cool script that may give an idea of how to optimize this process. Also carykh's project has another purpose, that is cutting with hand gestures and much other features.

**Comment by petermg** [comment]
> I found a script that does it much faster (by a lot) and doesnt have to extract each frame, saving time and space. Also it doesn't seem to have any A/V desync problems (an issue no one has reported yet).
> 
> https://github.com/excitoon/video-remove-silence
> 
> I am not trying to keep people away from carykh's project but to point to an alternative to just fullfill this task and show a script that may give an idea of how to optimize this process. Also carykh's project has another purpose, that is cutting with hand gestures and much more features.

The one you linked looks cool but has no tolerance settings so you cannot adjust it for different sensitivities.

**Comment by egrinant** [comment]
@petermg yes it has, look from line 17. Sorry for the formating, I am using the mobile. 

parser.add_argument('path', type=str, help='path to video')
parser.add_argument('--threshold-level', type=float, default=-40, help='threshold level in dB')
parser.add_argument('--threshold-duration', type=float, default=0.2, help='threshold duration in seconds')
parser.add_argument('--constant', type=float, default=0, help='duration constant transform value')
parser.add_argument('--sublinear', type=float, default=0, help='duration sublinear transform factor')
parser.add_argument('--linear', type=float, default=0.1, help='duration linear transform factor')
parser.add_argument('--save-silence', type=str, help='filename for saving silence')
parser.add_argument('--recalculate-time-in-description', type=str, help='path to text file')
